### PR TITLE
Remove FilePath

### PR DESCRIPTION
**Pull request by hawkowl**
This removes FilePath usage except where it's required by the Twisted API.

Refs #584 

cc @oberstet

### CODE REVIEW

**Comment by singlecheeze** [comment]
@hawkowl :two_hearts:

File: crossbar/worker/test/test_router.py [hunk 1]
```diff
@@ -337,7 +338,7 @@ def test_root_not_required(self):
         def done(results):
             for item in results:
                 if not item[0]:
-                    raise item[1]
+                    return item[1]
```
Comment (meejah) [inline crossbar/worker/test/test_router.py]: Was this raise->return intentional, as part of this "remove FilePath" stuff...?

**Comment by meejah** [comment]
LGTM, besides my one question inline in the diff...

**Comment by singlecheeze** [comment]
Bump for requested merge please after meejah's question is answered!

File: crossbar/worker/test/test_router.py (hunk 1 above)
Comment (hawkowl) [inline crossbar/worker/test/test_router.py]: Yeah, it was a test that didn't work on Py3 (I tried to run the tests as much as possible to make sure it didn't break anything)

**Comment by singlecheeze** [comment]
@hawkowl @meejah @markope 

```text
2016-01-13T19:16:31-0500 [Router       3048] Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1478, in dataReceived
    finishCallback(data[contentLength:])
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1736, in _finishRequestBody
    self.allContentReceived()
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1799, in allContentReceived
    req.requestReceived(command, path, version)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 832, in requestReceived
    self.process()
--- <exception caught here> ---
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 183, in process
    self.render(resrc)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 234, in render
    body = resrc.render(self)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/usr/local/lib/python3.5/dist-packages/crossbar/twisted/resource.py", line 320, in render_POST
    fileTempDir = os.path.join(self._tempDirRoot, fileId)
  File "/usr/lib/python3.5/posixpath.py", line 89, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.5/genericpath.py", line 145, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
builtins.TypeError: Can't mix strings and bytes in path components

2016-01-13T19:16:31-0500 [Router       3048] Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1478, in dataReceived
    finishCallback(data[contentLength:])
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1736, in _finishRequestBody
    self.allContentReceived()
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1799, in allContentReceived
    req.requestReceived(command, path, version)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 832, in requestReceived
    self.process()
--- <exception caught here> ---
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 183, in process
    self.render(resrc)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 234, in render
    body = resrc.render(self)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/usr/local/lib/python3.5/dist-packages/crossbar/twisted/resource.py", line 320, in render_POST
    fileTempDir = os.path.join(self._tempDirRoot, fileId)
  File "/usr/lib/python3.5/posixpath.py", line 89, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.5/genericpath.py", line 145, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
builtins.TypeError: Can't mix strings and bytes in path components

2016-01-13T19:16:31-0500 [Router       3048] Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1478, in dataReceived
    finishCallback(data[contentLength:])
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1736, in _finishRequestBody
    self.allContentReceived()
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1799, in allContentReceived
    req.requestReceived(command, path, version)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 832, in requestReceived
    self.process()
--- <exception caught here> ---
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 183, in process
    self.render(resrc)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 234, in render
    body = resrc.render(self)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/usr/local/lib/python3.5/dist-packages/crossbar/twisted/resource.py", line 320, in render_POST
    fileTempDir = os.path.join(self._tempDirRoot, fileId)
  File "/usr/lib/python3.5/posixpath.py", line 89, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.5/genericpath.py", line 145, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
builtins.TypeError: Can't mix strings and bytes in path components

2016-01-13T19:16:31-0500 [Router       3048] Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1478, in dataReceived
    finishCallback(data[contentLength:])
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1736, in _finishRequestBody
    self.allContentReceived()
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 1799, in allContentReceived
    req.requestReceived(command, path, version)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/http.py", line 832, in requestReceived
    self.process()
--- <exception caught here> ---
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 183, in process
    self.render(resrc)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/server.py", line 234, in render
    body = resrc.render(self)
  File "/usr/local/lib/python3.5/dist-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/usr/local/lib/python3.5/dist-packages/crossbar/twisted/resource.py", line 320, in render_POST
    fileTempDir = os.path.join(self._tempDirRoot, fileId)
  File "/usr/lib/python3.5/posixpath.py", line 89, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.5/genericpath.py", line 145, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
builtins.TypeError: Can't mix strings and bytes in path components
```

**Comment by hawkowl** [comment]
And this is why I like `FilePath`, since I specifically wrote `FilePath`'s Unicode support to handle this http://twistedmatrix.com/trac/ticket/7805 ;) 

Please file a new issue and I'll try and fix it.

**Comment by singlecheeze** [comment]
@hawkowl you do awesome work!

**Pull request merged**